url8867e6a4e2_cyberpython_JAwesomeChart_tgz-pJ8-com_gmigdos_jawesomechart_renderers_twodimensional_horizontal_ColumnChartRendererJ8/src/com/gmigdos/jawesomechart/core/Legend.java

```diff
@@ -0,0 +1,313 @@
+import com.gmigdos.jawesomechart.util.Utilities;
+import java.awt.*;
+import java.awt.geom.Rectangle2D;
+import java.util.Iterator;
+
+/**
+ *
+ * @author Georgios Migdos <cyberpython@gmail.com>
+ */
+public class Legend {
+
+    public enum LegendPosition {
+
+        LEGEND_POSITION_TOP, LEGEND_POSITION_BOTTOM, LEGEND_POSITION_LEFT, LEGEND_POSITION_RIGHT
+    };
+    private static final double DEFAULT_HORIZONTAL_GAP = 5.0;
+    private static final double DEFAULT_VERTICAL_GAP = 5.0;
+    private final static double DEFAULT_PADDING_VERTICAL = 10.0;
+    private final static double DEFAULT_PADDING_HORIZONTAL = 10.0;
+    private LegendPosition position;
+    private double height;
+    private double leftOrRightWidth;
+    private double topOrBottomWidth;
+    private DataSeriesList series;
+    private double horizontalGap;
+    private double verticalGap;
+    private double paddingVertical;
+    private double paddingHorizontal;
+    private Font font;
+    private Stroke legendBorder;
+    private Stroke markerBorder;
+    private Color legendBorderColor;
+    private Paint legendBackground;
+    private Color markerBorderColor;
+    private Color textColor;
+    private double markerDimension;
+    private Context2D context;
+
+    public Legend(double leftOrRightWidth, double topOrBottomWidth, DataSeriesList series) {
+        this(LegendPosition.LEGEND_POSITION_BOTTOM, leftOrRightWidth, topOrBottomWidth, series);
+    }
+
+    public Legend(Legend.LegendPosition position, double leftOrRightWidth, double topOrBottomWidth, DataSeriesList series) {
+        this.position = position;
+        this.series = series;
+        this.verticalGap = DEFAULT_VERTICAL_GAP;
+        this.horizontalGap = DEFAULT_HORIZONTAL_GAP;
+        this.paddingVertical = DEFAULT_PADDING_VERTICAL;
+        this.paddingHorizontal = DEFAULT_PADDING_HORIZONTAL;
+        this.font = new Font("Sans Serif", Font.BOLD, 10);
+        this.legendBorder = new BasicStroke();
+        this.markerBorder = new BasicStroke();
+        this.context = null;
+        this.legendBorderColor = new Color(153, 153, 153);
+        this.legendBackground = Color.WHITE;
+        this.markerBorderColor = Color.BLACK;
+        this.textColor = Color.BLACK;
+        this.leftOrRightWidth = leftOrRightWidth;
+        this.topOrBottomWidth = topOrBottomWidth;
+        this.markerDimension = 0;
+        updateSize();
+    }
+
+    public Context2D getContext() {
+        return context;
+    }
+
+    public void setContext(Context2D context) {
+        this.context = context;
+        updateSize();
+    }
+
+    public LegendPosition getPosition() {
+        return position;
+    }
+
+    public void setPosition(LegendPosition position) {
+        this.position = position;
+    }
+    
+    public boolean isPlacedOnLeftOrRight(){
+        return (position.equals(LegendPosition.LEGEND_POSITION_LEFT) || position.equals(LegendPosition.LEGEND_POSITION_RIGHT));
+    }
+
+    private void updateSize() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 44 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.